test/jdk/java/net/httpclient/offline/FixedResponseHttpClient.java

```diff
@@ -246,6 +249,13 @@ public HttpClient.Version version() {
      */
     private static class ConsumingSubscriber implements Flow.Subscriber<ByteBuffer> {
         final List<ByteBuffer> buffers = Collections.synchronizedList(new ArrayList<>());
+        // A CompletableFuture that will be completed with a list of ByteBuffers that the
+        // ConsumingSubscriber has consumed.
+        final CompletableFuture<List<ByteBuffer>> consumed = new CompletableFuture<>();
+
+        public final CompletableFuture<List<ByteBuffer>> getBuffers() {
+            return consumed;
+        }
```

Comment:
Is this method really needed? Above you can replace `subscriber.getBuffers().join()` with `subscriber.consumed.join()` too.

Reply:
I could have. I preferred calling a method though.